.circleci/config.yml

```diff
@@ -1,39 +1,114 @@
-version: 2
-jobs:
-  test:
+---
+version: 2.1
+
+executors:
+  # Common container definition used by all jobs
+  ruby_browsers:
     docker:
       - image: circleci/ruby:2.5.5-browsers
+
+commands:
+  # Smoke test command template
+  smoketest:
+    parameters:
+      env:
+        type: string
     steps:
       - checkout
       - run:
           name: Setup bundler
           command: gem install bundler
-      - restore-cache:
+      - restore_cache:
           keys:
             - v2-identity-monitor-bundle-{{ checksum "Gemfile.lock" }}
       - run:
           name: Install dependencies
           command: |
             bundle install --path vendor/bundle
-      - save-cache:
+      - save_cache:
           key: v2-identity-monitor-bundle-{{ checksum "Gemfile.lock" }}
           paths:
             - vendor/bundle
       - run:
           name: Setup tests
           command: bin/circleci-setup
       - run:
-          name: Run tests
-          command: bin/circleci-run
+          name: Run smoketest suite against environment
+          command: bin/circleci-run << parameters.env >>
+
+jobs:
+  smoketest_dev:
+    executor: ruby_browsers
+    steps:
+      - smoketest:
+          env: dev
+
+  smoketest_int:
+    executor: ruby_browsers
+    steps:
+      - smoketest:
+          env: int
+
+  smoketest_staging:
+    executor: ruby_browsers
+    steps:
+      - smoketest:
+          env: staging
+
+  smoketest_prod:
+    executor: ruby_browsers
+    steps:
+      - smoketest:
+          env: prod
+
 workflows:
-   version: 2
-   hourly:
-     triggers:
-       - schedule:
-           cron: "0 * * * *"
-           filters:
-             branches:
-               only:
-                 - master
-     jobs:
-       - test
+  # DEV - Not yet ready
+  # smoketest_dev:
+  #   triggers:
+  #     - schedule:
+  #         cron: "20 * * * *"
+  #         filters:
+  #           branches:
+  #             only:
+  #               - master
+  #   jobs:
+  #     - prepare
+  #     - smoketest_dev:
+  #       requires:
+  #         - prepare
+
+  # INT
+  smoketest_int:
+    triggers:
+      - schedule:
+          cron: "30 * * * *"
+          filters:
+            branches:
+              only:
+                - master
```

Comment:
It may make sense to have a branch for each env. Something that matches the IdP code. These frequently break because they are looking for a button label that is deployed to one env but not the other.

Reply:
Excellent idea!   Going to merge this now and if no one else beats me to it look into that later in the week.

Reply:
I think the per-branch stuff might work best if these lived in the same repo as the IDP